### PR TITLE
Feature: Added a setting to configure the "Open in ide" button on the status bar

### PR DESCRIPTION
**Pull request by yaira2**
**Resolved / Related Issues**

To prevent extra work, all changes to the Files codebase must link to an approved issue marked as `Ready to build`. Please insert the issue number following the hashtag with the issue number that this Pull Request resolves.
- Related to #12857

**Steps used to test these changes**

Stability is a top priority for Files and all changes are required to go through testing before being merged into the repo. Please include a list of steps that you used to test this PR.

1. Confirmed the "Open" button on the status bar is shown correctly based on the selected setting
2. Confirmed the "Open repo" option only shows for Git repos
3. Confirmed the status bar design and margins are the same as `main`

### CODE REVIEW

**Comment by yaira2** [comment]
@marcelwgn I discovered the issue and fixed it in [a2341d0](https://github.com/files-community/Files/pull/15394/commits/a2341d04036d7799c8c1cc20296d70cbe36744d7)

**Comment by yaira2** [comment]
I found two issues
- the folder item count isn't displayed
- the git actions are missing

File: src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs [hunk 1]
```diff
@@ -80,6 +81,14 @@ public string ExtendedStatusInfo
 			set => SetProperty(ref _ExtendedStatusInfo, value);
 		}
 
+		public bool ShowOpenInIDEButton
+		{
+			get
+			{
+				return DevToolsSettingsService.OpenInIDEOption == OpenInIDEOption.AllLocations || (DevToolsSettingsService.OpenInIDEOption == OpenInIDEOption.GitRepos && GitBranchDisplayName is not null);
```
Comment (0x5bfa) [inline src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs:88]: Can this be multiple lines?

File: src/Files.App/Services/Settings/IDevToolsSettingsService.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+using Microsoft.UI.Xaml;
+using Microsoft.UI.Xaml.Media;
+
+namespace Files.App.Services.Settings
+{
+	public interface IDevToolsSettingsService : IBaseSettingsService, INotifyPropertyChanged
+	{
+		/// <summary>
+		/// Gets or sets a value when the open in ide button should be displayed on the status bar.
```
Comment (0x5bfa) [inline src/Files.App/Services/Settings/IDevToolsSettingsService.cs:12]:
```suggestion
		/// Gets or sets a value when the open in IDE button should be displayed on the status bar.
```

File: src/Files.App/ViewModels/Settings/DevToolsViewModel.cs [hunk 3]
```diff
@@ -20,26 +22,44 @@ public bool IsLogoutEnabled
 			set => SetProperty(ref _IsLogoutEnabled, value);
 		}
 
-		public GitViewModel()
+		public DevToolsViewModel()
 		{
+			// Open in ide options
```
Comment (0x5bfa) [inline src/Files.App/ViewModels/Settings/DevToolsViewModel.cs:27]:
```suggestion
			// Open in IDE options
```

**Review by 0x5bfa — COMMENTED** [review]
Looks good to me code wise.

File: src/Files.App/UserControls/StatusBar.xaml [hunk 4]
```diff
@@ -193,6 +195,7 @@
 							ToolTipService.ToolTip="{x:Bind Commands.OpenInVSCode.LabelWithHotKey, Mode=OneWay}" />
 						<MenuFlyoutItem
 							x:Name="OpenRepoInVSCodeButton"
+							x:Load="{x:Bind DirectoryPropertiesViewModel.GitBranchDisplayName, Mode=OneWay, Converter={StaticResource NullToFalseConverter}, FallbackValue=False}"
```
Comment (hishitetsu) [inline src/Files.App/UserControls/StatusBar.xaml:198]: Nested `x:Load` may cause crashes, so it is better to use `Visiblity` instead.
https://github.com/files-community/Files/pull/12726#discussion_r1241172473

File: src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs:88]: How do you want it formatted?

File: src/Files.App/UserControls/StatusBar.xaml (hunk 4 above)
Comment (yaira2) [reply src/Files.App/UserControls/StatusBar.xaml]: Fixed

File: src/Files.App/UserControls/StatusBar.xaml [hunk 5]
```diff
@@ -196,7 +198,8 @@
 							Command="{x:Bind Commands.OpenRepoInVSCode}"
 							KeyboardAcceleratorTextOverride="{x:Bind Commands.OpenRepoInVSCode.HotKeys, Mode=OneWay}"
 							Text="{x:Bind Commands.OpenRepoInVSCode.Label, Mode=OneWay}"
-							ToolTipService.ToolTip="{x:Bind Commands.OpenRepoInVSCode.LabelWithHotKey, Mode=OneWay}" />
+							ToolTipService.ToolTip="{x:Bind Commands.OpenRepoInVSCode.LabelWithHotKey, Mode=OneWay}"
+							Visibility="{x:Bind DirectoryPropertiesViewModel.GitBranchDisplayName, Mode=OneWay, Converter={StaticResource NullToFalseConverter}, FallbackValue=False}" />
```
Comment (hishitetsu) [inline src/Files.App/UserControls/StatusBar.xaml:202]:
```suggestion
							Visibility="{x:Bind DirectoryPropertiesViewModel.GitBranchDisplayName, Mode=OneWay, Converter={StaticResource NullToVisibilityCollapsedConverter}, FallbackValue=Collapsed}" />
```

Comment (yaira2) [reply src/Files.App/UserControls/StatusBar.xaml:202]: Good catch, I should have noticed that.

File: src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs (hunk 1 above)
Comment (yaira2) [reply src/Files.App/Data/Models/DirectoryPropertiesViewModel.cs]: Done

**Review by 0x5bfa — APPROVED** [review]
Everything what I would do for readability and maintainability is in! LGTM, code wise.

**Pull request merged**